src/utils/baseError.ts

```diff
@@ -1,12 +1,14 @@
 class BaseError extends Error {
-  constructor(name: string, message: string) {
+  generic_error_code: string;
```

Comment:
Let's use field name like
- `identifier`
- `reason`
- `type`

Or something that will be generic and can fit in a single word. 

The purpose of the field will be extensively documented so it would be a one time learning that it is a platform agnostic error code.